src/js/core.js

```diff
@@ -1088,6 +1138,66 @@
                 target.innerHTML = html;
                 this.events.updateInput(target, { target: target, currentTarget: target });
             }
+        },
+
+        addElements: function (elements) {
+            var filtered = [];
+
+            // We want always an array with our input
+            if (!elements.length) {
+                elements = [elements];
+            }
+
+            // Filter the input, we want to include every element only once!
+            elements.forEach(function (element) {
+                if (element.getAttribute('data-medium-editor-element')) {
+                    return;
+                }
+
+                filtered.push(element);
+            });
+
+            // Do we have elements to add now?
+            if (filtered.length === 0) {
+                return false;
+            }
+
+            // Add new elements to our internal elements array
+            this.elements = this.elements.concat(filtered);
+
+            // Initialize all new elements (we check that in those functions don't worry)
+            initElements.call(this);
+            reAttachHandlers.call(this);
+        },
+
+        cleanupElements: function () {
```

Comment:
Totally agree - i'm refactoring the code accordingly i would say:

* remove cleanupElements
* remove the new function `findParentElementByTagName`
* add new function `removeElement`
* add new function `removeElements`
* add new function `addElement`

After that we're even - there are two fn's to add, two to remove with the same naming pattern. So it is also clear which one will take an element and which one an array as input ;)

Newly, the remove functionality will only check for the given "medium-editor-index" attribute to find his internal elements to match the ones we want to remove. So it is completely up to the implementation how one will do his "existence check" or, if you want to directly remove it on your "remove" hook :)